File: sol/2017/2017.py
class Solution:
    def gridGame(self, grid: List[List[int]]) -> int:
        n = len(grid[0])
        right = sum(grid[0]) - grid[0][0]
        res = right
        left = grid[1][0]
        for i in range(1, n):
            right -= grid[0][i]
            res = min(res, max(left, right))
            left += grid[1][i]
        return res

File: sol/2017/2017.cpp
struct Solution {
    using ll = long long;
    long long gridGame(vector<vector<int>>& grid) {
        int n = grid[0].size();
        ll right = 0, left =  grid[1][0];
        for (int i = 1; i < n; ++i){
            right += grid[0][i];
        }
        ll res = right;
        for (int i = 1; i < n; ++i){
            right -= grid[0][i];
            res = min(res, max(left, right));
            left += grid[1][i];
        }
        return res;
    }
};
